Rubberduck.VBEEditor/ComManagement/TypeLibs/TypeLibs.cs

```diff
@@ -0,0 +1,1669 @@
+        /// </summary>
+        /// <param name="controlName">the name of the control</param>
+        /// <returns>TypeInfoWrapper representing the type of control</returns>
+        public TypeInfoWrapper GetControlType(string controlName)
+        {
+            // TODO should encapsulate handling of raw datatypes
+            foreach (var func in Funcs)
+            {
+                using (func)
+                {
+                    // Controls are exposed as getters on the interface.
+                    //     can either be    ControlType* get_ControlName()       
+                    //     or               HRESULT get_ControlName(ControlType** Out) 
+
+                    if ((func.Name == controlName) &&
+                        (func.ProcKind == TypeInfoFunc.PROCKIND.PROCKIND_GET) &&
+                        (func.ParamCount == 0) &&
+                        (func.FuncDesc.elemdescFunc.tdesc.vt == (short)VarEnum.VT_PTR))
+                    {
+                        var retValElement = StructHelper.ReadStructure<ComTypes.ELEMDESC>(func.FuncDesc.elemdescFunc.tdesc.lpValue);
+                        if (retValElement.tdesc.vt == (short)VarEnum.VT_USERDEFINED)
+                        {
+                            return GetSafeRefTypeInfo((int)retValElement.tdesc.lpValue);
+                        }
+                    }
+                    else if ((func.Name == controlName) &&
+                        (func.ProcKind == TypeInfoFunc.PROCKIND.PROCKIND_GET) &&
+                        (func.ParamCount == 1) &&
+                        (func.FuncDesc.elemdescFunc.tdesc.vt == (short)VarEnum.VT_HRESULT))
+                    {
+                        // Get details of the first argument
+                        var retValElementOuterPtr = StructHelper.ReadStructure<ComTypes.ELEMDESC>(func.FuncDesc.lprgelemdescParam);
+                        if (retValElementOuterPtr.tdesc.vt == (short)VarEnum.VT_PTR)
+                        {
+                            var retValElementInnerPtr = StructHelper.ReadStructure<ComTypes.ELEMDESC>(retValElementOuterPtr.tdesc.lpValue);
+                            if (retValElementInnerPtr.tdesc.vt == (short)VarEnum.VT_PTR)
+                            {
+                                var retValElement = StructHelper.ReadStructure<ComTypes.ELEMDESC>(retValElementInnerPtr.tdesc.lpValue);
+
+                                if (retValElement.tdesc.vt == (short)VarEnum.VT_USERDEFINED)
+                                {
+                                    return GetSafeRefTypeInfo((int)retValElement.tdesc.lpValue);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            throw new ArgumentException($"TypeInfoWrapper::GetControlType failed. '{controlName}' control not found.");
```

Comment:
IDK if this is appropriate - when a `Find` fails, I usually expect a `null` result, not an exception.

Reply:
I personallyI think the term 'GetControlType' implies you are expecting it to exist.

Reply:
.... I don't know what to say. I could have sworn there was a method named `Find` adjacent to this but reading the code again, you're correct that it's `GetControlType`. Disregard & apologies.

Reply:
no worries :+1: